Tasks/ServiceFabricDeploy/utilities.ps1

```diff
@@ -0,0 +1,114 @@
+function Assert-SingleItem
+{
+    Param (
+        $Items,
+        
+        [String]
+        $Pattern
+    )
```

Comment:
The comparisons here look fragile. I would recommend checking the comparisons differently, perhaps:
```powershell
if (@($Items).Length -gt 1) { [...] }
elseif ($Items -eq $null -or @($Items).Length -eq 0) { [...] }
```

IIRC PowerShell documentation is explicitly vague exactly what type is returned when capturing the result of a pipeline with multiple values. Whether it's System.Array, System.ArrayList, or some other similar type of collection is never really clear in PowerShell.

The way the books recommend to ensure the result is an array (using the term array loosely), is to wrap it in `@( )`. If it's already an array, then it's not wrapped again. If it's a single element, then it's wrapped in an array.

Part of the fragility here also is that you're taking a dependency on the task lib to return a System.Array from Find-VstsFiles.